src/LuaShip.cpp

```diff
@@ -243,6 +282,33 @@ static int l_ship_set_skin(lua_State *l)
 }
 
 /*
+ * Method: SetPattern
+ *
+ * Changes the pattern used for texturing the ship.
+ *
+ * > ship:SetPattern(num)
+ *
+ * Parameters:
+ *
+ *   num - the pattern number
+ *
+ * Example:
+ *
+ * > ship:SetPattern(5)
+ *
+ * Status:
+ *
+ *  experimental
+ */
+static int l_ship_set_pattern(lua_State *l)
+{
+	Ship *s = LuaObject<Ship>::CheckFromLua(1);
+	unsigned int num = lua_tointeger(l, 2);
```

Comment:
Would it be possible to do some parameter sanitization here, and call lua_error if the pattern is invalid? Pretty please!

Note that I'd settle for return false on failure, if it were a common pattern in our API. Sadly, it isn't just yet.